### ISSUE TITLE
Sniper Elite V2 (63380)

### ISSUE THREAD

**Issue by tfaller** [ValveSoftware/Proton#392]
The launcher of the game shows up, but the actual game can't be launched by the launcher.
There is nothing special in the steam-63380.log file.

OS: Ubuntu 18.04 64-bit
CPU: AMD Athlon(tm) II X4 620
GPU: AMD Radeon HD 7800 Series
Driver Version: Mesa 18.3.0-devel - padoka PPA

**Comment by Warrengbrn** [comment]
Same here.

Ubuntu 18.04.1
Ryzen 7 1700
Nvidia Geforce GTX 1070
Driver 354.51

**Comment by priand1** [comment]
Can confirm this. My System Information:
https://gist.github.com/FreigeistZ/df2db421094762a1a14e184585fd9eee

The logfile of appid 63380:

[steam-63380.log](https://github.com/ValveSoftware/Proton/files/2349228/steam-63380.log)

**Comment by IanTrudel** [comment]
I managed to launch Sniper Elite V2 using Lutris and Winesteam.

Running the game requires the following:
- Update GNU TLS in the appropriate Steam directory by removing and symlink libgnutls.so.26 to the newer version.
- Install Xact using winetricks or protontricks if you have troubles with sound.
- Refresh rate might be a problem. Useful command line options:
  - Use `-<refresh rate>Hz` such as `-60Hz`, `-70Hz`, etc.
  - Otherwise use `-subwindow` and press `ALT-ENTER`.

**Comment by priand1** [comment]
This is on Proton 4.2-3, my gist is two comments above.
Sniper Elite V2 is still crashing when pressing the play button in the launcher.

In the attached log you can see that Proton is trying to load
C:\\Program Files (x86)\\Steam\\Steam2.dll
as a native dll which does not work, because it does not exist.

After that it tries to load 
/Steam/steamapps/common/Proton 4.2/dist/lib/wine/steam2.dll.so
as built-in dll which also fails because it also does not exist.

SniperEliteV2.exe terminates after that

FWIW, I found a native Steam2.dll in a Steam for Windows installation and copied it over to the
/Steam/steamapps/compatdata/63380/pfx/drive_c/Program Files (x86)/Steam/
folder and the game still crashes. I deleted that native dll afterwards.

After that I found all files in
/Steam/steamapps/compatdata/63380/pfx/drive_c/Program Files (x86)/Steam/
namely
385312 Apr 22 19:01 steamclient64.dll
385312 Apr 22 19:01 steamclient.dll
385312 Apr 22 19:01 Steam.dll
share the same size and md5sum.

So I copied Steam.dll as Steam2.dll into this folder which resulted in the game starting but hanging with a black screen.
[steam-63380.log](https://github.com/ValveSoftware/Proton/files/3104244/steam-63380.log)

**Comment by kisak-valve** [comment]
Sniper ELite V2 Doesn't launcher!

Issue transferred from https://github.com/ValveSoftware/Proton/issues/4676.
@AndyMxM posted on 2021-03-17T21:26:33:

# Compatibility Report
Sniceper Elite V2 Doens't launch
- Steam AppID of the game: 63380

## System Information
- GPU: GTX 1050
- Driver/LLVM version: NVIDIA-DIRVER 460(propietary)
- Kernel version: 5.4.0-67-generic
- Link to full system information report as [Gist](https://gist.github.com/)
-  (https://gist.github.com/AndyMxM/a1fc7e26ed0f73dbbae9597b1c817731#file-steam-63380-log)
- 
- Proton version: 5.13-6

## I confirm:
- [x ] that I haven't found an existing compatibility report for this game.
- [ x] that I have checked whether there are updates for my system available.


(https://gist.github.com/AndyMxM/a1fc7e26ed0f73dbbae9597b1c817731#file-steam-63380-log)


## Symptoms 
I am trying to launch the game Sniper Elite V2, but it trying to load but doesn't work, and even it doesn't show me any error message!

## Reproduction

**Comment by NoXPhasma** [comment]
With Steam Beta + Proton Experimental it shows the launcher but crashes when I press play in the launcher.
https://gist.github.com/NoXPhasma/533c7562e7fb7366545517ca4a05ef67
[steam-63380.log](https://github.com/ValveSoftware/Proton/files/7441344/steam-63380.log)

**Comment by NoXPhasma** [comment]
Retested with Proton Experimental Bleeding Edge and it still crashes after the launcher.
[steam-63380.log](https://github.com/ValveSoftware/Proton/files/7446855/steam-63380.log)

**Comment by leillo1975** [comment]
7 years later the game don't start. It shows the launcher, but when you hit "play" on it, nothing happens

**Comment by Isaac-Clarke** [comment]
Sorry for the "me too" message, but:

Both on my brother's W10 pc and my current Fedora 43 PC the game won't launch pass the launcher.

I played this game in the past, I finished it. I have no idea what happened.

**Comment by Hoff16** [comment]
> Sorry for the "me too" message, but:
> 
> Both on my brother's W10 pc and my current Fedora 43 PC the game won't launch pass the launcher.
> 
> I played this game in the past, I finished it. I have no idea what happened.

The game works on Windows 11 for me but still crashes on Linux on the same machine after I press Play in the launcher.

Looks like it might still be this steam2.dll issue that is being tracked here:

https://github.com/ValveSoftware/Proton/issues/2613

**Comment by Hoff16** [comment]
Also just so we have a recent Proton log:

My System Info:
https://gist.github.com/Hoff16/e5bc6fd19f7f1d192f61edde4eab2e1b

Proton Log:

[steam-63380.log](https://github.com/user-attachments/files/29105325/steam-63380.log)

This was made using GE-Proton10-34

**Comment by Hoff16** [comment]
I've been tackling this and it seems that it might be related to CEG/DRM. Right before the SniperEliteV2.exe process terminates it loads and unloads the following cryptographic DLLs (crypt32.dll, advapi32.dll, rsaenh.dll).

- crypt32.dll appears briefly then gets unloaded
- advapi32.dll is referenced again
- rsaenh.dll loads, which is the RSA Enhanced Cryptographic Provider CSP

The fact that crypt32.dll unloads immediately after suggests the certificate operations are complete, then control returns to the higher-level crypto APIs in advapi32.dll. The trace then checks for mscoree.dll (the .NET CLR) right before terminating (a routine compatibility check that many applications perform) and immediately exits with code -1, indicating a deliberate failure.

The final sequence shows the process loading the cryptographic stack (crypt32.dll, advapi32.dll, and rsaenh.dll for RSA operations) and then terminating. The crypto sequence points to a failure in RSA or certificate-based verification (likely CEG/DRM ticket validation, Steam ownership checks, or copy-protection integrity) that either crashes outright under Wine's CryptoAPI layer or technically succeeds but returns unexpected results that trigger the deliberate exit.

@kisak-valve is there anything that I can provide to assist in troubleshooting this further? Just want to get visibility here as this game has never launched properly under Proton.

**Comment by Hoff16** [comment]
Some more testing has revealed that this was a red herring. Given that earlier solutions on ProtonDB were tied to a set of modified files (exe included). I figured that I would dig into that deeper and see what those files were doing.

It turns out those files were potentially evading Steam CEG DRM and so that could be the culprit here. I am going to continue testing but, I was under the impression that CEG DRM on Steam was fully resolved many years ago. Hopefully this can be looked into by the powers-that-be at Valve to come up with an out-of-the-box fix!

I'll keep digging and try to isolate exactly what those replacement files are doing.

EDIT: Removed references to legally questionable links.

**Comment by kisak-valve** [comment]
Hello @Hoff16, friendly reminder that I'm a moderator for these issue trackers and not a developer myself. The link you posted is legally problematic and has been removed.

**Comment by Hoff16** [comment]
> Hello @Hoff16, friendly reminder that I'm a moderator for these issue trackers and not a developer myself. The link you posted is legally problematic and has been removed.

Apologies and note! I'm just enthusiastic to get one of my last Windows-only games fixed!

**Comment by simifor** [comment]
@Holf16 can you try with proton bleeding edge? https://github.com/ValveSoftware/Proton/wiki/Proton-Versions#proton-bleeding-edge

**Comment by Hoff16** [comment]
> @Holf16 can you try with proton bleeding edge? https://github.com/ValveSoftware/Proton/wiki/Proton-Versions#proton-bleeding-edge

Uninstalled the game and reinstalled with Proton Experimental Bleeding Edge and that launcher doesn't even appear, let alone the game.

**Comment by simifor** [comment]
@Hoff16 first verify game files and if the issue remains try attaching a new proton log

**Comment by Hoff16** [comment]
Verified game files and confirmed that bleeding edge was selected. Here is a fresh new Proton log.

[steam-63380.log](https://github.com/user-attachments/files/30139943/steam-63380.log)

Based on past experiences here and in ProtonDB this appears to be a regression at some point as the launcher previously opened!

**Comment by kisak-valve** [comment]
Friendly note that the log you shared isn't using the Proton build you expected.

**Comment by Hoff16** [comment]
Confirmed that Proton 6.3-8, 7.0-6, 8.0-5, 9.0-4, &10.0-4, all show the Launcher.

Starting with Proton 11.0 and now Experimental (Bleeding Edge). The Launcher no longer appears, please mark this as a Regression if necessary!

Note: I deleted the prefix in CompatData and Verified game files in between each Proton version change.

Thank you.

**Comment by Hoff16** [comment]
> Friendly note that the log you shared isn't using the Proton build you expected.

Thanks @kisak-valve , I see that now. No new Proton Log is even generated using Proton Experimental (Bleeding Edge). The game simply moves from Play -> Launching -> Play very quickly. Not sure how to properly log this!

**Comment by kisak-valve** [comment]
That hints that something went awry before Proton had a chance to start. I'd completely close Steam, then run `steam` from a terminal and check the terminal spew around when you tell Steam to start the game. Alternatively, you can monitor `<steam install folder>/logs/console_log.txt` for the same terminal spew.

**Comment by Hoff16** [comment]
> That hints that something went awry before Proton had a chance to start. I'd completely close Steam, then run `steam` from a terminal and check the terminal spew around when you tell Steam to start the game. Alternatively, you can monitor `<steam install folder>/logs/console_log.txt` for the same terminal spew.

@kisak-valve Here is the terminal spew from both launching steam and launching the game/the game crashing before getting to the launcher:

https://gist.github.com/Hoff16/d9057d1a5d99b4046d8ea1f2c61d50ab

**Comment by kisak-valve** [comment]
`sh: line 1: /home/hoff/Steam Library 2/SteamLibrary/steamapps/common/SteamLinuxRuntime_4/_v2-entry-point: Permission denied` is the line to focus on. I'd double check how the partition with the secondary library is mounted, in particular if it's noexec or if the regular user has permission issues there.

**Comment by Hoff16** [comment]
> `sh: line 1: /home/hoff/Steam Library 2/SteamLibrary/steamapps/common/SteamLinuxRuntime_4/_v2-entry-point: Permission denied` is the line to focus on. I'd double check how the partition with the secondary library is mounted, in particular if it's noexec or if the regular user has permission issues there.

Ok, resolved that error by modifying fstab as that mount point was set to noexec.

Now on Proton Experimental (Bleeding Edge) the game launches!!

Thank you @simifor & @kisak-valve for your help! Especially kisak for putting up with my shenanigans! <3

**Comment by Hoff16** [comment]
FYI All, this commit in Bleeding Edge Proton on July 16th is why the game is working now!

http://github.com/ValveSoftware/wine/commit/91e9e8

**Comment by Hoff16** [comment]
Also, looks like even on Stable (11.0) and Experimental simply setting the launch option: 

WINESTEAMNOEXEC=1 %command%

The game will launch and play perfectly!

**Comment by Isaac-Clarke** [comment]
I have tested WINESTEAMNOEXEC=1 %command% , but since I have been experiencing too many variables on my PC (I updated CPU, RAM & MOBO and [one of my drives went](https://discussion.fedoraproject.org/t/mounting-permanently-a-storage-unit-in-fedora-kde-automount-at-boot-no-password-all-users-can-see-and-edit-files/148030/40) "linux steam library folder is not executable") to say "this actually worked".

I now don't even have that command on, I only have Proton Experimental and it works. :/

**Comment by Hoff16** [comment]
> I have tested WINESTEAMNOEXEC=1 %command% , but since I have been experiencing too many variables on my PC (I updated CPU, RAM & MOBO and [one of my drives went](https://discussion.fedoraproject.org/t/mounting-permanently-a-storage-unit-in-fedora-kde-automount-at-boot-no-password-all-users-can-see-and-edit-files/148030/40) "linux steam library folder is not executable") to say "this actually worked".
> 
> I now don't even have that command on, I only have Proton Experimental and it works. :/

@Isaac-Clarke  Are you on Bleeding Edge?

That command is only needed if you aren't on Bleeding Edge Experimental.

Eventually it will move upstream to regular Experimental and then stable Proton.

**Comment by Isaac-Clarke** [comment]
Sorry for the late answer @Hoff16 , but I had... things...

Yes, I am on Bleeding Edge because otherwise Helldivers 2 doesn't launch on Linux (they did something with the antivirus and "fullscreen detection" or whatever).

---

Edit:
I changed to B.E. BECAUSE of Helldivers 2. Most of all other games are on GE34 or whatever the latest is right now.

I've been using GE because of NTsync, which is now present on Proton 11, so I may just never get a new GE installed if it isn't either for Heroic or for singular games with special cases.

**Comment by TechniKris** [comment]
Indeed, setting `WINESTEAMNOEXEC=1` or using Proton Experimental Bleeding Edge does fix the issue for me as well! 🎉
At first the game only loaded up to the Sniper Elite V2 logo splash, with all the copyright information at the bottom and closed before displaying a single frame of the main menu. If that happens, you might need to verify your game files from within Steam (I ended up uninstalling and reinstalling the entire game while troubleshooting). 

Also, something else worth noting: after launching the game in the Rebellion launcher the game will hang at a black screen for 30 seconds, before flashing white and then continuing startup with the Rebellion logo animation. It seems to happen every time. I'm not sure if it's an issue on my end, so for reference: I'm on an Nvidia 2070, and I tried with both the propriety (nvidia-open) 610 series driver as well as NVK from Mesa 26.1.5. 
Edit: Also, alt-tabbing out and back in makes the game ignore all m/kb inputs and needs to be restarted (I haven't tried gamepad yet)

**Comment by Hoff16** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/392#issuecomment-5084146490

@TechniKris Fantastic that it works for you also! Feel free to make a post on ProtonDB so we can raise the rating there!

Mine hangs as well at startup (I have an Intel iGPU so not an Nvidia issue)

Mine also ignores input after alt-tabbing but I got it to work somehow and then after that could alt+tab as much as I wanted in that session. It was purely mouse and keyboard also.

**Comment by leillo1975** [comment]
Finally I could play this game wit experimental. Thanks a lot! . But I found two problems:
-Alt+Tab makes the game unresponsible
-Sound has a lot of echo, which is particularly noticeable in the voices of the German soldiers and over the loudspeakers on the battlefield

**Comment by Isaac-Clarke** [comment]
> -Alt+Tab makes the game unresponsible

For this one, maybe, just maybe, you should try using GameScope.
The easiest way to set it is by getting Steam Thinker Launch from ProtonUp-Qt.

**Comment by Hoff16** [comment]
> > -Alt+Tab makes the game unresponsible
> 
> For this one, maybe, just maybe, you should try using GameScope.
> The easiest way to set it is by getting Steam Thinker Launch from ProtonUp-Qt.

Great suggestion! I'll test this out in a few hours!

**Comment by Hoff16** [comment]
@Isaac-Clarke & @leillo1975 gamescope seems to solve the problem but it kinda just avoids it in a way by making its own gamescope window.

Also you lose out on Steam Overlay and any features related to it which is a known issue it seems. Still at least its a workaround for the issue in the meantime!

**Comment by Hoff16** [comment]
Also update! The game (as of today!) now works out-of-the-box with Experimental! No need for Bleeding Edge or a Launch Command!

**Comment by Isaac-Clarke** [comment]
> Also you lose out on Steam Overlay

Gamescope comes with Steam.
It's the default on Steam Deck and the Steam Machine 2 because it makes it easier to control the game in Big Picture mode.

**Comment by Hoff16** [comment]
> > Also you lose out on Steam Overlay
> 
> Gamescope comes with Steam.
> It's the default on Steam Deck and the Steam Machine 2 because it makes it easier to control the game in Big Picture mode.

When I launch gamescope and then attempt to launch Steam overlay nothing happens, I also don't have the built-in fps counter etc in the top right. Is that expected behavior with gamescope @Isaac-Clarke ?

**Comment by Isaac-Clarke** [comment]
Now I've tried it out on my own @Hoff16 , since I don't usually play games using GameScope, and I must say that Steam Overlay, at least like this, indeed doesn't seem to work.

(I used Far Cry 2 because I already have it set up for that.)

That said, the performance overlay can be exchanged with mangoapp (or just MangoHUD on its own, with GOverlay or whatever).

tl;dr too bad, but unless a Distro adds a better way to handle windows, or Steam makes GameScope better (unless there's a command I don't know about) the Steam Overlay is a good tradeoff compared to not be able to alt-tab...

**Comment by Hoff16** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/392#issuecomment-5098268812

@Isaac-Clarke 

Well I'm hoping that we can figure out the root cause of the issue and get the game to be able to regain focus without having to resort to using gamescope!

I agree that:
Alt+Tab Support > Steam Overlay

**Comment by Hoff16** [comment]
Hi All,
  I believe I've found the cause of the 30-second wait after pressing Play in the Launcher:

SniperEliteV2.exe initializes Steam’s DRM IPC channel:

```
STEAM_DIPC_CONSUME
SREAM_DIPC_PRODUCE
shared mapping STEAM_DRM_IPC
```


It then creates an acknowledgment event named:

`STEAM_START_ACK_EVENT_664_4342052`

That event is assigned handle 0x120. The game writes its .STEAMSTART notification, releases the IPC semaphore, and waits on that acknowledgment event with a 29.999-second timeout.

The acknowledgment never arrives.



Exactly 29.999 seconds later, Wine returns:

```
NtWaitForSingleObject -> 0x102

0x102 is STATUS_TIMEOUT. The game then closes the Steam DRM IPC handles and immediately begins DXVK initialization.
```



So the actual sequence is:

```
SniperEliteV2.exe starts
↓
Creates Steam DRM IPC objects
↓
Sends .STEAMSTART notification
↓
Waits for STEAM_START_ACK_EVENT
↓
No acknowledgment arrives
↓
29.999-second timeout
↓
Steam DRM IPC is abandoned
↓
DXVK initializes
↓
White flash
↓
Rebellion logo
```

Basically this means that it is, once again, a Valve CEG DRM issue. There are a smattering of reports of other CEG DRM games with this 30-second black screen after launching the game so I guess this is what's happening here.

Not sure what the other games are doing but hopefully Valve/CodeWeavers can see this and implement some sort of fix for us!!

**Comment by simifor** [comment]
The game should be launching faster on bleeding edge

**Comment by Hoff16** [comment]
> The game should be launching faster on bleeding edge

@simifor that's great news! I will confirm this in a few hours!

**Comment by Hoff16** [comment]
> The game should be launching faster on bleeding edge

Amazing! Zero waiting time now before launch! Thank you @simifor 

Here is the commit that fixed it:

https://github.com/ValveSoftware/Proton/commit/ab36742586370bf6fc6c7397415dd9e4404523ed

Shoutout to Paul Gofman at CodeWeavers for both this bug and getting Sniper Elite V2 to launch at all through Proton!

**Comment by simifor** [comment]
It was Plagman's doing, I'm just reporting it  :)

**Comment by Hoff16** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/392#issuecomment-5084146490

@TechniKris the 30 second black screen hang has been fixed in Proton Experimental Bleeding Edge!

Should be in the next Experimental release which should come by the end of August.

**Comment by Hoff16** [comment]
FYI all as of August 18th, 2026 the Experimental build now includes the fix for the 29.999 second black screen before the Rebellion logo.

`CEG-using games should no longer take extra time to start.`

Source: https://github.com/ValveSoftware/Proton/wiki/Changelog#available-in-proton-experimental-as-of-2026-08-18

@TechniKris

**Comment by Hoff16** [comment]
FYI all as of August 21st, 2026, Proton 11.0-2 includes the fix for the 29.999 second black screen before the Rebellion logo!

`CEG-using games should no longer take extra time to start.`

Looks like the startup fix has not been pushed to Stable yet so the game will not launch without the Launch Command:

`WINESTEAMNOEXEC=1 %command%`

EDIT: I was having an issue replicating the Alt+Tab input issue we were all seeing. It's possible there was change that resolved it as I wasn't seeing it on Proton 9/10/11 as long as I had the `WINESTEAMNOEXEC=1` launch command.

However, it's also possible that we were so set on troubleshooting the game launching and the hang at startup that we neglected to properly test older Proton versions for the Alt+Tab issue.

Therefore it might be a regression in Experimental/now 11.0-2 but I need to do some more testing to be sure.

**Comment by TechniKris** [comment]
The alt+tab input issue still occurs for me, on both Experimental and stable (11.0-2 as of now).
But aside from that: I confirm that with Experimental the game launches flawlessly; with 11.0-2 it still requires the `WINESTEAMNOEXEC` envvar, but launches without the half minute timeout.

**Comment by Hoff16** [comment]
I am seeing the same thing @TechniKris, it seems that this Alt+Tab issue is actually a regression that we never noticed.

With Proton 10.0-4 I do not see the issue and its possible that it was working with Proton 11.0-1. I will do some testing with specific commits to see if I can narrow down the issue to a specific commit made during that time that may have caused it.

**Comment by Hoff16** [comment]
<summary>
Apologies for the incoming wall of text but here is a summary tests I've run over the course of the past month regarding this issue.
</summary>

See Below:

<details>

Sniper Elite V2: Fullscreen Alt+Tab Causes Complete Input Loss

Game: Sniper Elite V2
Steam AppID: 63380
Environment: KDE Plasma Wayland/XWayland

REGRESSION

Proton 10.0-4: WORKS
Proton 11.0-2: BROKEN
Proton Experimental: BROKEN

The issue was present in Proton Experimental before the release of Proton 11.0-2 and now appears to have propagated into stable Proton 11.0-2.

PROBLEM

When Sniper Elite V2 is running fullscreen under affected Proton versions, switching away from the game and returning causes the game to stop accepting all keyboard and mouse input.

Basic reproduction:

1. Launch Sniper Elite V2.
2. Enter fullscreen mode.
3. Verify keyboard and mouse work.
4. Alt+Tab away from the game.
5. Return to the game.
6. Keyboard and mouse no longer work.

The failure is deterministic and can be reproduced repeatedly.

Explicitly minimizing the fullscreen game through KWin and restoring it causes the same input failure, so the problem is not specific to the Alt+Tab keyboard combination. It appears to be associated with losing/restoring foreground state while fullscreen.


RECOVERY BEHAVIOR

Pressing Alt+Enter after input has broken immediately restores input by switching the game to windowed mode.

The game can then be switched back to fullscreen with Alt+Enter and input continues working normally.

However, another fullscreen Alt+Tab causes the failure again:

Fullscreen + working
        ↓
Alt+Tab out/in
        ↓
Keyboard + mouse dead
        ↓
Alt+Enter
        ↓
Windowed + input restored
        ↓
Alt+Enter
        ↓
Fullscreen + still working
        ↓
Alt+Tab out/in
        ↓
Keyboard + mouse dead again

This strongly suggests that the display/window-mode transition resets whatever input/application state becomes stuck during fullscreen deactivation/reactivation.


GAMESCOPE

Running Sniper Elite V2 through Gamescope completely prevents the problem.

Repeated Alt+Tab cycles work without losing keyboard or mouse input.


WINE VIRTUAL DESKTOP

Enabling:

winecfg -> Graphics -> Emulate a virtual desktop

also completely prevents the problem.

This was confirmed while using DXVK, so the result is not an artifact of falling back to WineD3D.

Both Gamescope and Wine virtual desktop change how the game's window interacts with the host compositor/window manager.


DXVK VS WINED3D

The issue was tested with normal Proton/DXVK:

Proton + DXVK
-> BROKEN

It was then tested with:

PROTON_USE_WINED3D=1 %command%

Result:

Proton + WineD3D
-> BROKEN

Switching from DXVK to WineD3D therefore does not eliminate the problem, making the graphics translation backend unlikely to be the determining factor.


WINE DEBUGGING

Wine debug logging was collected around successful input, focus loss, focus restoration, and failed input.

On fullscreen Alt+Tab, Wine appears to perform the expected focus/activation sequence.

On focus loss, messages include:

WM_ACTIVATE
WM_ACTIVATEAPP(FALSE)
WM_KILLFOCUS

When returning to the game, X11/Wine reports focus restoration, including:

WM_TAKE_FOCUS
set_focus setting foreground window
set_focus setting focus
FocusIn
X11DRV_FocusIn

Wine then delivers the expected Win32 activation messages:

WM_ACTIVATEAPP(TRUE)
WM_ACTIVATE
WM_SETFOCUS

Therefore, this does not appear to be a simple case of Wine failing to return focus to the game's window.


DIRECTINPUT BEHAVIOR

DirectInput tracing provided a more interesting distinction.

When fullscreen focus is lost, Wine unacquires the foreground DirectInput device as expected.

After the broken fullscreen return:

Focus restored
WM_ACTIVATEAPP(TRUE)
WM_ACTIVATE
WM_SETFOCUS
        ↓
SEV2 does not resume its normal DirectInput polling/reacquisition
        ↓
Keyboard + mouse remain dead

During the working Alt+Enter/window-mode transition, the game resumes DirectInput activity, including calls such as:

GetDeviceData()
Acquire()
GetDeviceData()
GetDeviceState()

Input then immediately works again.

This suggests Wine successfully restores window activation, but SEV2 does not resume its input-processing/reacquisition path following the affected fullscreen foreground-loss cycle.


WINDOWED BEHAVIOR

Using Alt+Enter to put the game into windowed mode avoids the input-loss behavior.

Windowed behavior itself is somewhat unusual: the game tends to remain visually above other windows rather than behaving like a conventional window.

The working windowed traces also showed a somewhat different window lifecycle, including WM_SIZE/window-position activity.


EXPLICIT MINIMIZE TEST

To determine whether Alt+Tab itself was the trigger, the fullscreen game was explicitly minimized using KWin and then restored.

Result:

Fullscreen + working
-> explicitly minimize
-> restore
-> input dead

Therefore, the failure is not specific to Alt+Tab.

A fullscreen minimize/deactivate and restore cycle is sufficient to reproduce it.


WINDOWS COMPARISON

Sniper Elite V2 was also tested under Windows running through WinBoat.

The game's fullscreen behavior is unusual there as well. After Alt+Tab, another application receives focus, but SEV2 can initially remain visually in front. Typing then causes SEV2 to minimize and exposes the selected application.

Despite that quirk, repeated cycles of leaving SEV2 and returning to it do not cause input loss under Windows.

Keyboard and mouse continue working every time.

Because WinBoat uses virtualization/remote presentation, this is not considered equivalent to a bare-metal Windows control, but it demonstrates that Windows can repeatedly deactivate/restore the game without producing the stuck-input state seen under affected Proton versions.


UPSTREAM WINE TEST

Sniper Elite V2 was also tested using a separate upstream Wine 11.15-devel installation running the Windows Steam client.

The game successfully launched and fullscreen Alt+Tab did not cause input loss.

Graphics were not rendering correctly in this configuration, so this is not considered a perfectly equivalent comparison. However, it provides additional evidence that the failure may be specific to Proton/Valve's Wine environment rather than generic upstream Wine behavior.


PROTON VERSION TESTING

The most important version comparison so far is:

Proton 10.0-4:
WORKS

Proton 11.0-2:
BROKEN

Current Proton Experimental:
BROKEN

Proton Experimental exhibited the issue before the release of 11.0-2, and the same behavior now appears in stable Proton 11.0-2.

This suggests that a change previously present in Experimental may have propagated into Proton 11.0-2.


CURRENT WORKING HYPOTHESIS

The evidence currently points toward a fullscreen foreground/minimize/restore interaction in Proton/Valve Wine rather than a general keyboard/mouse problem or a DXVK-specific problem.

In affected Proton versions:

Fullscreen game
        ↓
Foreground lost/minimized
        ↓
Foreground DirectInput state is unacquired
        ↓
Window restored
        ↓
Wine reports FocusIn
WM_ACTIVATEAPP(TRUE)
WM_ACTIVATE
WM_SETFOCUS
        ↓
SEV2 does not resume its DirectInput polling/reacquisition path
        ↓
Keyboard + mouse remain nonfunctional

A display/window-mode transition with Alt+Enter causes the game to resume its input path immediately.

Gamescope and Wine virtual desktop both prevent the bad state from occurring.


CURRENT TEST MATRIX

Proton 10.0-4, fullscreen:
WORKS

Proton 11.0-2, fullscreen:
BROKEN

Current Proton Experimental, fullscreen:
BROKEN

Experimental + DXVK:
BROKEN

Experimental + WineD3D:
BROKEN

Experimental + Gamescope:
WORKS

Experimental + Wine virtual desktop + DXVK:
WORKS

Experimental windowed via Alt+Enter:
WORKS

Alt+Enter after input failure:
Immediately restores input

Return to fullscreen after Alt+Enter:
Continues working

Next fullscreen Alt+Tab:
Breaks again

Explicit fullscreen minimize/restore:
Input lost

Upstream Wine 11.15-devel:
Alt+Tab input survives

Windows via WinBoat:
Input survives repeated focus-loss cycles


LIKELY NEXT STEP

Given that Proton 10.0-4 works while Proton 11.0-2 and Experimental fail, the most useful next step is probably identifying the Wine/Proton change between the known-good and known-bad versions that altered fullscreen foreground/minimize/restore behavior.

The renderer has already been isolated as unlikely to be responsible, and the problem appears reproducible enough that testing a candidate revert should provide a clear result.

</details>

**Comment by Hoff16** [comment]
@simifor any reason why this commit wasn't pushed to stable?

https://github.com/ValveSoftware/wine/commit/91e9e8fbf62074ade996b167010dd27bf6a90d98

**Comment by codeweaverwill** [comment]
@Hoff16 @TechniKris Thanks all for the good info and persistence. If you get a chance, could you try Proton Experimental Bleeding Edge for the alt+tab input loss issue? There's a fix in BE that works on my end.

**Comment by Hoff16** [comment]
> [@Hoff16](https://github.com/Hoff16) [@TechniKris](https://github.com/TechniKris) Thanks all for the good info and persistence. If you get a chance, could you try Proton Experimental Bleeding Edge for the alt+tab input loss issue? There's a fix in BE that works on my end.

@codeweaverwill 
Amazing! Alt+Tab issue is gone as well as the previous issues with CEG causing the game to fail to start and the 30-second delay.

I think we are at feature-parity with Windows now (on Bleeding Edge) Thank you!!

**Comment by Hoff16** [comment]
FYI All, we are now at feature-parity with Windows on regular Experimental as of the release today (8/28/26)! Game starts after the launcher, right to the Rebellion logo (no 30 sec delay), and can be successfully Alt+Tabbed without issue!

Waiting with bated breath for these changes to propagate upstream to Stable Proton!
